StrongInject.Generator/Visitors/DependencyCheckerVisitor.cs

```diff
@@ -146,7 +146,7 @@ protected override bool ShouldVisitAfterUpdateState(InstanceSource source, State
         
         protected override void AfterVisit(InstanceSource source, State state)
         {
-            if (source is { IsAsync: true, Scope: not Scope.SingleInstance } and not DelegateSource && !state.IsScopeAsync
+            if (source is { IsAsync: true, Scope: not Scope.SingleInstance } and not (DelegateSource or OwnedSource) && !state.IsScopeAsync
```

Comment:
Possible future PR to refactor this so that there meaning of `IsAsync` can be understood without knowing about DelegateSource and OwnedSource. (https://github.com/YairHalberstadt/stronginject/pull/143#discussion_r689031075)